### PR TITLE
remove version requirement

### PR DESCRIPTION
**Pull request by mkangia**
Since we swapped over HQ to start protecting the Drift session variable callout such that it now works if the variable isn't there in https://github.com/dimagi/commcare-hq/pull/21098, we can also remove the requirement that the HQ App version is 2.44 which was originally added in https://github.com/dimagi/commcare-hq/pull/20757

### CODE REVIEW

File: corehq/apps/app_manager/xform.py [hunk 1]
```diff
@@ -1360,14 +1358,6 @@ def add_meta_2(self, form):
...
         )
 
-        if add_meta_drift:
-            self.add_setvalue(
-                ref="meta/drift",
-                event="xforms-revalidate",
-                value="if(count(instance('commcaresession')/session/context/drift) = 1, "
-                      "instance('commcaresession')/session/context/drift, '')",
-            )
-
```
Comment (esoergel) [inline corehq/apps/app_manager/xform.py]: I'm confused - isn't this removing the drift expression entirely?  This removes all of the drift related code from both of those PRs.

**Review by esoergel — COMMENTED** [review]
Build failure looks real:
```
======================================================================
FAIL: corehq.apps.app_manager.tests.test_form_preparation_v2:TestFormMeta.test_meta_drift
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/test_form_preparation_v2.py", line 603, in test_meta_drift
    self.form.render_xform()
AssertionError: False is not true
```

File: corehq/apps/app_manager/xform.py (hunk 1 above)
Comment (mkangia) [reply corehq/apps/app_manager/xform.py]: I apologize. Did all that fairly quickly and incorrectly. I meant to just remove the if condition. Added back meta.

Comment (esoergel) [reply corehq/apps/app_manager/xform.py]: Ah cool, this makes more sense.

**Review by esoergel — APPROVED** [review]
Looks like this will require a bunch of random unrelated tests to be updated, since they do an exact match on xforms :(

**Comment by mkangia** [comment]
Thanks @esoergel.

Fixed all.

**Review by esoergel — APPROVED** [review]

**Pull request merged**